vault/request_handling.go

```diff
@@ -584,6 +584,32 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 		return nil, nil, ctErr
 	}
 
+	// If this is a wrapping request, check for token validity before routing
+	switch req.Path {
+	case "sys/wrapping/rewrap", "sys/wrapping/unwrap":
```

Comment:
Don't we have to consider the case of `sys/wrapping/wrap` anywhere?

Reply:
`sys/wrapping/wrap` does not contain a provided wrapping token, so there's nothing to validate. It will simply go through the normal request flow, and perform audit log that way.

Reply:
Doesn't it need `sys/wrapping/lookup` though?

Reply:
Are you referring to `sys/wrapping/wrap` needing `sys/wrapping/lookup`, or the switch case itself? If it's the latter, it's handled in https://github.com/hashicorp/vault/pull/6541/files#diff-90984b74207af54c7b459b3c29bb3fefR983

Reply:
Token validation for `sys/wrapping/lookup` is handled in `handleLoginRequest`, marking this as resolved.